x-pack/plugins/endpoint/public/plugin.ts

```diff
@@ -9,12 +9,14 @@ import { EmbeddableSetup } from 'src/plugins/embeddable/public';
 import { DataPublicPluginStart } from 'src/plugins/data/public';
 import { i18n } from '@kbn/i18n';
 import { ResolverEmbeddableFactory } from './embeddables/resolver';
+import { IngestManagerSetup } from '../../ingest_manager/public';
```

Comment:
Up until now I have been consuming Ingest APIs via REST - see file under `x-pack/plugins/endpoint/public/applications/endpoint/services/ingest.ts`.
I'm wondering if we should remain consistent until all services are exported from ingest?

Reply:
Oh I didn't even realize we had that. I talked with John Schulz about just using the REST APIs and he suggested using the platform's dependency sharing paradigm instead (having a plugin return a contract that other plugins can call) which is what I've used here. I think this would make it easier to transition those calls to the inter-plugin contract.